### PR TITLE
[state:modified] persist unrendered_config from schema.yml, and more reliably compute unrendered_config from .sql files

### PR DESCRIPTION
**Pull request by MichelleArk**
Resolves https://github.com/dbt-labs/dbt-core/issues/9564

### Problem

1) A node's unrendered_config included rendered values, because the population of unrendered_config from models with config in .sql files was occurring after kwargs were evaluated by value (e.g. materialized=(jinja expression) => jinja expression evaluated by the time we are in ConfigProviders)

2) We were not storing the unrendered_configs from schema.yml parsing

### Solution

All behind the new behaviour flag, `state_modified_compare_more_unrendered_values`
1) Parse the unrendered_config on the node from .sql from the raw model code and store values as string representations of jinja calls
   * multiple config calls are merged naively, since we only have string types in the values
2) Store unrendered_config on the schema file under manifest.file
   * storage under schema.yml key (e.g. "models") -> name (versioned names have _v{version} suffixes
3) During node parsing, look up the unrendered_configs from the schema file and use them as the `patch_config_dict`

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me.
- [x] I have run this code in development, and it appears to resolve the stated issue.
- [x] This PR includes tests, or tests are not required or relevant for this PR.
- [x] This PR has no interface changes (e.g., macros, CLI, logs, JSON artifacts, config files, adapter interface, etc.) or this PR has already received feedback and approval from Product or DX.
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions.

### CODE REVIEW

**Comment by MichelleArk** [comment]
Corresponding schemas PR: https://github.com/dbt-labs/schemas.getdbt.com/pull/60

File: core/dbt/artifacts/resources/v1/components.py [hunk 1]
```diff
@@ -194,13 +194,16 @@ class ParsedResource(ParsedResourceMandatory):
...
+    unrendered_config_call_dict: Dict[str, Any] = field(default_factory=dict)
     relation_name: Optional[str] = None
     raw_code: str = ""
 
     def __post_serialize__(self, dct: Dict, context: Optional[Dict] = None):
         dct = super().__post_serialize__(dct, context)
         if context and context.get("artifact") and "config_call_dict" in dct:
             del dct["config_call_dict"]
+        if context and context.get("artifact") and "unrendered_config_call_dict" in dct:
+            del dct["unrendered_config_call_dict"]
```
Comment (MichelleArk) [inline core/dbt/artifacts/resources/v1/components.py:206]: even though this isn't serialized, we still need a schemas.getdbt.com PR, similar to config_call_dict: https://github.com/dbt-labs/schemas.getdbt.com/pull/60

File: core/dbt/context/providers.py [hunk 2]
```diff
@@ -395,6 +396,13 @@ def __call__(self, *args, **kwargs):
         # not call it!
         if self.context_config is None:
             raise DbtRuntimeError("At parse time, did not receive a context config")
+
+        # Track unrendered opts to build parsed node unrendered_config later on
```
Comment (MichelleArk) [inline core/dbt/context/providers.py:400]: TODO: put this call behind behaviour flag as well

File: core/dbt/parser/base.py [hunk 3]
```diff
@@ -369,13 +373,27 @@ def update_parsed_node_config(
...
+                        "models", parsed_node.name, getattr(parsed_node, "version", None)
+                    ):
+                        patch_config_dict = deep_merge(patch_config_dict, unrendered_patch_config)
+
         # unrendered_config is used to compare the original database/schema/alias
         # values and to handle 'same_config' and 'same_contents' calls
         parsed_node.unrendered_config = config.build_config_dict(
             rendered=False, patch_config_dict=patch_config_dict
         )
+        print(f"final unrendered_config: {parsed_node.unrendered_config}")
```
Comment (MichelleArk) [inline core/dbt/parser/base.py:393]: oops

File: core/dbt/parser/schemas.py [hunk 4]
```diff
@@ -464,6 +485,8 @@ def parse(self) -> ParseResult:
                 self.manifest.source_patches[key] = patch
                 source_file.source_patches.append(key)
             else:
+                # TODO: add unrendered_database from manifest.unrendered_source_patch
```
Comment (MichelleArk) [inline core/dbt/parser/schemas.py:488]: in follow-up PR

**Review by gshank — APPROVED** [review]
It would be really nice to spend some time making all the config application and merging code cleaner. Maybe next quarter...

**Pull request merged**